### ISSUE TITLE
Проблема с локализацией и ее решение

### ISSUE THREAD

**Issue by SUWAROW** [BasilYes/godot-yandex-games-sdk#25]
```gdscript
func _ready() -> void:
	Yandex.init_game()
	await Yandex.game_initialized
	TranslationServer.set_locale(Yandex.lang)
```

Необходимо указать, что функция set_locale может быть исполнена только в корневой сцене иначе ничего не получится. См. документацию Godot.
Также, необходима задержка, если игра запущена, то меняй язык.

**Comment by BasilYes** [comment]
Я не знаю, зачем тут что-то менять. У меня в проекте не в яндекс играх спокойно меняется перевод на лету не из корневой сцены. Зачем ждать когда игра запуститься чтобы поменять язык? В документации ничего не нашел про это
![image](https://github.com/user-attachments/assets/bfe2aa9a-59fa-450a-92d5-1d07f8c0a441)

**Comment by BasilYes** [comment]
Зачем? Вызов смены языка буквально находиться перед вызовом сигнала
```gdscript
func _game_initialized(args) -> void:
	app_id = args[0].app.id
	lang = args[0].i18n.lang
	tld = args[0].i18n.tld
	payload = args[0].payload
	is_game_initialized = true
	TranslationServer.set_locale(lang) # Вот вызов set_locate
	game_initialized.emit() #  Вот вызов сигнала
```

Про корневую сцену ничего не нашел в документации, в другом проекте делал делал set_locale из синглтона, все работало прекрастно на лету
![image](https://github.com/user-attachments/assets/d9a2a3b3-3ee2-458f-a9be-e3465710b3dc)

**Comment by SUWAROW** [comment]
> Зачем? Вызов смены языка буквально находиться перед вызовом сигнала
> 
> ```gdscript
> func _game_initialized(args) -> void:
> 	app_id = args[0].app.id
> 	lang = args[0].i18n.lang
> 	tld = args[0].i18n.tld
> 	payload = args[0].payload
> 	is_game_initialized = true
> 	TranslationServer.set_locale(lang) # Вот вызов set_locate
> 	game_initialized.emit() #  Вот вызов сигнала
> ```
> 
> Про корневую сцену ничего не нашел в документации, в другом проекте делал делал set_locale из синглтона, все работало прекрастно на лету ![image](https://private-user-images.githubusercontent.com/36816595/367612915-d9a2a3b3-3ee2-458f-a9be-e3465710b3dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5MjM1MTQsIm5iZiI6MTcyNjkyMzIxNCwicGF0aCI6Ii8zNjgxNjU5NS8zNjc2MTI5MTUtZDlhMmEzYjMtM2VlMi00NThmLWE5YmUtZTM0NjU3MTBiM2RjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIxVDEyNTMzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlmNjk2OWYzODYzZTI4YTAwZGFmNTU1YWRmY2EyMGRhMmVhYTNkNWVlZDNkOTY4NGY1MmM3MzJjMThiNjFlNzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rDABj6MDTk5b94ZUW0pQH6chidnCBfGV7_XSwmKl5TQ)

Уточни на какой версии Godot ты ведешь разработку?

**Comment by BasilYes** [comment]
4.3

**Comment by SUWAROW** [comment]
> 4.3

https://github.com/godotengine/godot/issues/85051

Короче, эта проблема была и якобы она не повторяется на 4.3. У тебя 4.3. релизная или предыдущие?
Или может что-то нужно сделать помимо описанного тобой в readme.md?

**Comment by BasilYes** [comment]
Добавил функцию уже на 4.3. В любом случае, чтобы билд для яндекса сделать нужен 4.3

**Comment by BasilYes** [comment]
Проблема решена или нет? Могу закрывать?

**Comment by SUWAROW** [comment]
Нет, не получилось. Мы ушли с плагина на добавление СДК в <head>.
